webrtc.html

```diff
@@ -8774,6 +8774,18 @@ <h2>
                   privacy-sensitive contexts, browsers can consider mitigations
                   such as reporting only a common subset of the capabilities.
                 </p>
+                <p class="no-test-needed">
+                  Prior to calling {{RTCPeerConnection/createOffer}} or
+                  {{RTCPeerConnection/createAnswer}} there MAY be capabilities
+                  that are either unknown when the synchronous
+                  {{getCapabilities()}} call is made, or that the user agent
+                  chooses not to disclose for other reasons (e.g. privacy). If
+                  {{getCapabilities()}} is called after
+                  {{RTCPeerConnection/createOffer}} or
+                  {{RTCPeerConnection/createAnswer}} has resolved, the
+                  capabilities returned by this method MUST match that of the
+                  most recently created offer or answer.
```

Comment:
Note that before this PR it is spec-compliant to implement getCapabilities() as "return {};". After this PR, that is not spec-compliant.

To address the concern about getCapabilities() being static, but offers and answers are on a per-peer connection basis, how about instead of saying "capabilities returned by this method MUST match that of the most recently created offer or answer" it can say "capabilities returned by this method MUST match _or be a superset_ of _any previously_ created offer or answer _by any peer connections in the document_".

1. This is consistent with the method's description that this is an "optimistic view" of the capabilities.
2. Calling getCapabilities() after calling createOffer() could theoretically add to the list of the capabilities previously returned, but it could _never subtract from it_.
3. Because you cannot have capabilities exposed in SDP that are not also included in getCapabilities(), this PR means you no longer* have to parse SDP to get a full set of capabilities that your peer connection exposes.
4. In practise, no browser has to update their implementation.

Today's fingerprinting note **already** makes getCapabilities() a foot-gun, but this PR makes it slightly less of a foot-gun by adding some requirement to what it MUST return.

_* In practise of course you don't have to do this, but a browser could return a subset of capabilties from what is exposed in SDP and still be compliant._

Tagging @jan-ivar @youennf for thoughts but this can wait until the interim, no rush

Reply:
With regards to 2), I guess the "optimistic view" of capabilities would allow you to actually be over-optimistic before createOffer(), then do createOffer() and find out you don't have a HW encoder, and then remove that from the set of capabilities. But my point is you should never have something in the SDP that is not also in the a later getCapabilities() call.